src/main/java/com/amplitude/HttpTransport.java

```diff
@@ -311,11 +311,14 @@ protected boolean shouldRetryForStatus(Status status) {
   }
 
   private void triggerEventCallbacks(List<Event> events, int status, String message) {
-    if (callbacks == null || events == null || events.isEmpty()) {
+    if (events == null || events.isEmpty()) {
       return;
     }
     for (Event event : events) {
-      callbacks.onLogEventServerResponse(event, status, message);
+      AmplitudeCallbacks eventCallback = event.callback != null ? event.callback : callbacks;
```

Comment:
I think should we still call the client level callbacks here in addition to the per-event callback

Reply:
We can do that too, I was trying to debate if we need to do replace or in addition. Would in addition sounds more natural for the intention?

Reply:
Updated to call extra now.